doc/neps/tools/build_index.py

```diff
@@ -0,0 +1,93 @@
+"""
+Scan the directory of nep files and extract their metadata.  The
+metadata is passed to Jinja for filling out `index.rst.tmpl`.
+"""
+
+import os
+import sys
+import jinja2
+import glob
+import re
+
+
+def render(tpl_path, context):
+    path, filename = os.path.split(tpl_path)
+    return jinja2.Environment(
+        loader=jinja2.FileSystemLoader(path or './')
+    ).get_template(filename).render(context)
+
+def nep_metadata():
+    ignore = ('nep-template.rst')
+    sources = sorted(glob.glob(r'nep-*.rst'))
+    sources = [s for s in sources if not s in ignore]
+
+    meta_re = r':([a-zA-Z\-]*): (.*)'
+
+    neps = {}
+    print('Loading metadata for:')
```

Comment:
Seems like we should be able to parse the `rst` file with `docutils`

Reply:
Indeed, it is possible, but would look  something like this: https://stackoverflow.com/questions/12883428/how-to-parse-restructuredtext-in-python